src/client/base.ts

```diff
@@ -0,0 +1,33 @@
+import { Aptos, Network } from '@aptos-labs/ts-sdk'
+import { Client } from 'urql'
+
+import {
+  defaultAptosClient,
+  defaultAptosGraphqlClient,
+  defaultMirageGraphqlClient,
+  MirageConfig,
+  mirageConfigFromNetwork,
+} from '../utils/config'
+
+export class MirageClientBase {
+  protected network: Network
+  protected aptosGraphqlClient: Client
+  protected mirageGraphqlClient: Client
+  protected aptosClient: Aptos
+  protected config: MirageConfig
+
+  constructor(
+    network: Network,
+    config?: MirageConfig,
+    aptosClient?: Aptos,
+    aptosGraphqlApiKey?: string,
+    aptosGraphqlClient?: Client,
+    mirageGraphqlClient?: Client,
```

Comment:
would it make more sense to just pass in the aptos config here and construct internally?